frontend/src/models/TagAssignee.ts

```diff
@@ -0,0 +1,6 @@
+import { UserName } from './User';
+
+export interface TagAssignee {
+  userName?: UserName;
```

Comment:
Nah I didn’t will go check

Reply:
This name here is solely for display purposes so id has no real use. Basically in the backed I’m not saving the user id and hydrating details largely because we aren’t dealing with a relational db and doing those kinds of join operations would be much slower in mongo. So instead

Reply:
I think we're going to run into problems if we use display names instead of IDs. For example if we have logic like `if assignee === me` and that's based only on name, then when two people have the same name it's gonna break. Storing userId+name+avatar in mongo wouldn't be hard, but I would've thought we just store the ID and then query github for the name/avatar/email/etc?

Reply:
LOL it is confusing. My simple feature and feature is not that bad compared to this.

Reply:
How about changing it to "assigneeName"? Clearer and tag-related. Also, why import UserName? Isn't it just a string type?

Reply:
Haha that might be a close second :P - one day I plan to rename SimpleFeature to FeatureSummary or something like that.

Reply:
UserName is an alias of string, which I made because we had maps like <string,string> and it was confusing what they were, so now we can say <UserName,whatever> and it reads nicer. BUT, it should be <UserId,whatever> not name.

Reply:
No!!! It takes me so long to come up with this name.

Reply:
Yep, we do have the logic when reassigning other's tag. if the tag is assigned to someone else previously, you will have a warning dialogue when reassigning. If two people have the same name, it won't break, but you won't have the warning dialogue.

Reply:
@kevinoshea we can't just query github as once we pull down the user credentials and store them we throw away the github token. I don't think we would want to add more github integration than login as then we have to store and manage tokens for all the rest calls we make. 

Also if we have say 15 users assigned to tags then we would have to make 15 rest calls to github just to figure out who is assigned to what. I feel like storing the display details is the cleanest solution to this.

You are right about the id thing though I will add that to the thing stored then I think we can just use the user class as most of the stuff in there is optional.

Reply:
You don't need a github token to access this info though, it's publicly accessible. I can get all your info (avatar/name/email/etc) from just your ID by hitting https://api.github.com/users/ryan-rushton and also your avatar is https://github.com/ryan-rushton.png
So we already know the avatar URL, and the frontend could query that user API when you hover over a user or something to show their details - full name, email, etc.

I'm not sure how other auth systems would work, but in the case of LDAP we could provide similar endpoints to retrieve this data as well.

Normally storing user details is not recommended because they would become stale (eg you could have two different avatars showing up at the same time, or an old email address, etc). I realise it's not a huge issue for us here but generally-speaking it would be an issue.

Reply:
You don't need a github token to access this info though, it's publicly accessible. I can get all your info (avatar/name/email/etc) from just your ID by hitting https://api.github.com/users/ryan-rushton and also your avatar is https://github.com/ryan-rushton.png
So we already know the avatar URL, and the frontend could query that user API when you hover over a user or something to show their details - full name, email, etc.

I'm not sure how other auth systems would work, but in the case of LDAP we could provide similar endpoints to retrieve this data as well.

Normally storing user details is not recommended because they would become stale (eg you could have two different avatars showing up at the same time, or an old email address, etc). I realise it's not a huge issue for us here but generally-speaking it would be an issue.

Reply:
You can't quite get everything, on going to mine you will notice that email is null as you need to be authenticated for that. I didn't realise that you could get the avatar that way so maybe we could just use the github login of the user to get it.

The only issue with using those is that login can change (its not the github id). For example my gitthub used to be rushton3179 and I changed it to ryan-rushton. We would then need to have a fall back in the event that a user changes their name and the overlap period has run out, or a user has deleted their account.

The only details I need for the avatar bit is a url to put as the source of the avatar and their name so that I can make initials.

Reply:
I was thinking the frontend would just fetch the avatar via an img src. Not sure why you'd need to store it? It'd get cached in the browser anyway. You don't necessarily need to use socialLogin because the backend github component could construct the avatarUrl from the ID and return that for the frontend to use as is.

Regarding the fact that email is hidden - that's a good thing I think. If your github preferences are to hide your email then I don't think we should store a copy of it and then expose it to other users. Fetching this info also means that you can turn hide permissions on/off whenever you want.

The ID changing is an issue, though a stored avatarUrl may not work either then. Unless github accounts have a static ID that we should be using instead? Deleted accounts are fine - they shouldn't show any details anyway.

Reply:
Ok that being said I will make some changes tonight and we can have a look at it after

Reply:
Another problem I just thought of - if we are doing logic like `if assignee == me` then we should really be using ID + sourceLogin as the identifier. Because the same ID could live on 2 different systems and be used by 2 different users. It's like ID and namespace.